src/mavlink/manager.rs

```diff
@@ -1 +1,203 @@
+            if let Err(error) = inner
+                .lock()
+                .unwrap()
+                .sender
+                .send(Message::Received((header, message)))
+            {
+                error!("Failed handling message: {error:?}");
+                continue;
+            }
+        }
+    }
+
+    #[instrument(level = "debug", skip(inner))]
+    fn sender_loop(inner: Arc<Mutex<Connection>>) {
+        let mut receiver = { inner.lock().unwrap().sender.subscribe() };
+
+        loop {
+            loop {
+                std::thread::sleep(std::time::Duration::from_micros(10));
+
+                // Receive answer from the cameras
+                let (header, message) = match receiver.try_recv() {
+                    Ok(Message::ToBeSent(message)) => message,
+                    Err(broadcast::error::TryRecvError::Closed) => {
+                        unreachable!(
+                            "Closed channel: This should never happen, this channel is static!"
+                        );
+                    }
+                    Ok(Message::Received(_)) | Err(_) => continue,
+                };
+
+                // Send the response from the cameras to the Mavlink network
+                if let Err(error) = inner.lock().unwrap().connection.send(&header, &message) {
+                    error!("Failed sending message to Mavlink Connection: {error:?}");
+                    break;
+                }
+
+                trace!("Message sent: {header:?}, {message:?}");
+            }
+
+            // Reconnects
+            {
+                let mut inner = inner.lock().unwrap();
+                inner.connection = Connection::connect(&inner.address);
+            }
+        }
+    }
+
+    #[instrument(level = "debug")]
+    pub fn new_component_id() -> u8 {
+        let manager = MANAGER.lock().unwrap();
+
+        let mut id = mavlink::common::MavComponent::MAV_COMP_ID_CAMERA as u8;
+        let mut vector = manager.ids.lock().unwrap();
+
+        // Find the closest ID available
+        while vector.contains(&id) {
+            id += 1;
+        }
+
+        vector.push(id);
+        id
+    }
+
+    #[instrument(level = "debug")]
+    pub fn drop_id(id: u8) {
+        let manager = MANAGER.lock().unwrap();
+        let mut vector = manager.ids.lock().unwrap();
+
+        if let Some(position) = vector.iter().position(|&vec_id| vec_id == id) {
+            vector.remove(position);
+        } else {
+            error!("Id not found");
+        }
+    }
+
+    #[instrument(level = "debug")]
+    pub fn get_sender() -> broadcast::Sender<Message> {
+        let manager = MANAGER.lock().unwrap();
+
+        let connection = manager.connection.lock().unwrap();
+
+        connection.sender.clone()
+    }
+}
+
+impl Connection {
+    #[instrument(level = "debug")]
+    fn connect(address: &str) -> Box<dyn MavConnection<MavMessage> + Sync + Send> {
+        debug!("Connecting...");
+
+        loop {
```

Comment:
```suggestion
        loop {
            debug!("Connecting to: {address}");
```

Reply:
Great, I'll put it inside the loop. 

Just so you know, when we add the tracing::instrument derive macro for a function, it instruments the function parameters in the log, so the address is already in the record :)
It will show like:
```
2023-09-14T16:24:05.013431Z DEBUG                 main ThreadId(01) connect{address="tcpout:0.0.0.0:5777"}: src/mavlink/manager.rs:188: Connecting...
```